core/src/main/java/org/renjin/primitives/subset/Subsetting.java

```diff
@@ -339,8 +343,58 @@ public static SEXP setSingleElement(SEXP source, @ArgumentList ListVector argume
       subscripts.add(argumentList.get(i));
     }
 
+
+
+
+    if (source instanceof ListVector && isRecursiveIndexingArgument(subscripts) ) {
+
+      SEXP result = source;
+
+      if (argumentList.get(0).length() == 3) {
+        int itr = source.getIndexByName(argumentList.get(0).getElementAsSEXP(0).asString());
```

Comment:
You've hardcoded the indexes here; can you try to use the `SelectionStrategy` class instead?